_posts/2018-01-03-Gauge-automated-testing.md

```diff
@@ -0,0 +1,286 @@
+* Show a message "Thank you for signing up! You are now logged in."
+```
+
+Cucumber only offers this feature in some implementations, they don't in Cucumber.js for example.
+They refuse to implement it because they believe this creates too much abstraction and makes you lose sight of the overall picture.
+Gauge does offer it, so it's up to you whether you want to make use of it or not.
+
+# Running the specs
+Since the project is setup with Maven, the tests can be run with `mvn test`.
+However, if you want to pass any arguments, you'll need to use `mvn gauge:execute` instead.
+
+### Tags
+
+You may have noticed in the spec files that tags can be added.
+They can be used to run only certain specs.
+
+`mvn gauge:execute -DspecsDir=specs -Dtags="sign-up & customer"`
+
+### Parallel
+So far, Gauge didn't stand out from other automated testing solutions.
+Although there's one thing that really does stand out and that's how easy it is to use parallel execution!
+
+`mvn gauge:execute -DspecsDir=specs -DinParallel=true`
+
+Running this command will look how many CPU cores your computer has and start up that many streams.
+For each stream it will open a browser window and execute the specs.
+So if you have four CPU cores, four browser windows will be opened.
+You can overwrite the number of parallel executors, but it's recommended not to exceed the number of CPU cores.
+
+`mvn gauge:execute -DspecsDir=specs -DinParallel=true -Dnodes=2`
+
+Now, some specs may take longer to run than others.
+By default specs are divided dynamically over the streams.
+So when a spec has finished, it will take the next one from the list of specs that still need to be executed.
+It's possible to change this so the specs are divided on startup, but the command is deprecated and will be removed.
+
+### Making tests independent
+To make use of this parallel execution, you'll have to make sure your tests don't rely on each other.
+I think it's pretty clear why you shouldn't do that.
+Anyway, suppose you test the sign up and sign in.
+If your sign in relies on the user being signed up through a previous test, these tests should be run synchronously.
+However, if you would want to test whether a user can sign in after having signed up, you should do so in one test.
+That immediately solves our problem and we are safe to use the parallel execution!
+
+# Environments and configuration
+The config files are located under `env/default`.
+You should have three files in that folder: `default.properties`, `java.properties` and `user.properties`.
+In the [example by Gauge](https://github.com/getgauge-examples/java-maven-selenium){:target="_blank"}, they have an `APP_URL` parameter in that last file.
+I recommend using that approach as well, you can get parameters in your Java code using `System.getenv("APP_URL")`.
+
+It's possible to create different environments by simply creating a new folder and add the configuration files that you want to change there.
```

Comment:
... and **adding** the ...
Not sure if I understand the last part of the sentence, "the configuration files that you want to change"?
You might mean specifying config files for multiple environment if certain properties differ depending on the environment?